### PR TITLE
Resolve more compile warnings

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: iothub/service/src/Common/Exceptions/ConfigurationNotFoundException.cs [hunk 1]
```diff
@@ -1,37 +1,70 @@
...
         {
         }
 
+        /// <summary>
+        /// For serialization purposes
+        /// </summary>
+        /// <param name="info">Serialization info</param>
+        /// <param name="context">Streaming context</param>
+#pragma warning disable CA2229 // Implement serialization constructors
+
```
Comment (barustum) [inline iothub/service/src/Common/Exceptions/ConfigurationNotFoundException.cs]: extra line

File: iothub/service/src/Common/ReadOnlyDictionary45.cs [hunk 2]
```diff
@@ -81,20 +103,36 @@ public ValueCollection Values
...
         }
 
         ICollection<TKey> IDictionary<TKey, TValue>.Keys => Keys;
 
+        /// <summary>
+        /// Gets the value of the specified key, if exists
+        /// </summary>
+        /// <param name="key"></param>
+        /// <param name="value"></param>
+        /// <returns></returns>
```
Comment (barustum) [inline iothub/service/src/Common/ReadOnlyDictionary45.cs]: Return description

File: iothub/service/src/Common/Exceptions/ConfigurationNotFoundException.cs (hunk 1 above)
Comment (drwill-ms) [reply iothub/service/src/Common/Exceptions/ConfigurationNotFoundException.cs]: VS saves it that way. /shrug

File: iothub/device/tests/Edge/EdgeModuleClientFactoryTest.cs [hunk 3]
```diff
@@ -101,7 +100,7 @@ public async Task TestCreate_FromEnvironment_MissingVariable_ShouldThrow()
         }
 
         [TestMethod]
-        public void TestCreate_FromEnvironment_UnsupportedAuth_ShouldThrow()
+        public async Task TestCreate_FromEnvironment_UnsupportedAuth_ShouldThrow()
```
[the remark this answers is not in the document]
Comment (drwill-ms) [reply iothub/device/tests/Edge/EdgeModuleClientFactoryTest.cs]: We don't do that for test methods.

File: common/src/service/HttpClientHelper.cs [hunk 4]
```diff
@@ -674,8 +686,12 @@ private static void InsertEtag(HttpRequestMessage requestMessage, string etag)
             IDictionary<HttpStatusCode, Func<HttpResponseMessage, Task<Exception>>> errorMappingOverrides,
             CancellationToken cancellationToken)
         {
-            using (var httpClient = BuildHttpClient(_defaultOperationTimeout))
+            HttpClient httpClient = null;
+            HttpClientHandler httpClientHandler = null;
+
+            try
```
Comment (vinagesh) [inline common/src/service/HttpClientHelper.cs]: I should have probably pointed this out too :) Same suggestion as above for using.

File: common/src/service/HttpClientHelper.cs [hunk 5]
```diff
@@ -700,19 +721,34 @@ private static void InsertEtag(HttpRequestMessage requestMessage, string etag)
...
-                await ExecuteAsync(
-                    httpClient,
-                    httpMethod,
-                    requestUri,
-                    modifyRequestMessageAsync,
-                    isMappedToException,
-                    processResponseMessageAsync,
-                    errorMappingOverrides,
-                    linkedCts.Token).ConfigureAwait(false);
+                (httpClient, httpClientHandler) = BuildHttpClient(Timeout.InfiniteTimeSpan);
```
Comment (vinagesh) [inline common/src/service/HttpClientHelper.cs]: I should have probably pointed this out too :) Same suggestion as above for using.

**Pull request merged**